drivers/include/dose.h

```diff
@@ -153,7 +154,7 @@ typedef struct {
     dose_state_t state;                     /**< Current state of the driver's state machine */
     mutex_t state_mtx;                      /**< Is unlocked every time a state is (re)entered */
     uint8_t recv_buf[DOSE_FRAME_LEN];       /**< Receive buffer for incoming frames */
-    size_t recv_buf_ptr;                    /**< Index of the next empty octet of the recveive buffer */
+    chunk_ringbuf_t rb;                     /**< Ringbuffer to store received frames */
```

Comment:
```suggestion
    chunk_ringbuf_t rb;                     /**< Ringbuffer to store received frames.       */
                                            /* Written to from interrupts (with irq_disable */
                                            /* to prevent any simultaneous writes),         */
                                            /* consumed exclusively in the network stack's  */
                                            /* loop at _isr.                                */
```

(Should have been part of the full review which I hope to continue tomorrow, got interrupted).